scout/adapter/mongo/query.py

```diff
@@ -756,6 +756,26 @@ def secondary_query(self, query, mongo_query, secondary_filter=None):
                 mongo_secondary_query.append({"cosmic_ids": {"$exists": True}})
                 mongo_secondary_query.append({"cosmic_ids": {"$ne": None}})
 
+            if criterion == "fusion_score":
+                mongo_secondary_query.append(
+                    {"fusion_score": {"$gte": float(query.get("fusion_score"))}}
+                )
+            if criterion == "ffpm":
+                mongo_secondary_query.append({"samples.0.ffpm": {"$gte": float(query.get("ffpm"))}})
+            if criterion == "junction_reads":
+                mongo_secondary_query.append(
+                    {"samples.0.read_depth": {"$gte": float(query.get("junction_reads"))}}
```

Comment:
I don't find the `junction_reads` anywhere in the scout code? Are you sure it is currently parsed from the VCF? What's the VCF field name?

Reply:
juction reads is the read_depth is the "DV" value in the vcf

Reply:
it goes into the field `read_depth`

Reply:
Ah OK, but it's not called like that in the variant, so the filter won't work

Reply:
what RD is not given in the fusion vcf

Reply:
> but the filter does work, which field to take from the mongo db is given on `samples.0.read_depth`

Ah yes, you are right, I got confused, sorry!

Reply:
we want to show DV (junction reads) and RV (split reads) separate. They give different information and I would not add those values

Reply:
> we want to show DV (junction reads) and RV (split reads) separate. They give different information and I would not add those values

I understand, but be aware that whenever ref_depth is -1 the it is computed from those 2 values together, see [here](https://github.com/Clinical-Genomics/scout/blob/a95fb214586f87f31460273beb58324b1f500297/scout/parse/variant/genotype.py#L326)

Reply:
I see, but it should not be -1 once the bug is fixed, right?

Reply:
Other question: for junction reads you mean the paired ends right? Why not naming them like that?

Reply:
these are the reads spanning the junction of the fusion gene. I think paired reads is too vague